### ISSUE TITLE
Sending heartbeat from within a Job

### ISSUE THREAD

**Issue by davechallis** [rq/rq#805]
I'm wondering if it's possible to trigger the currently executing Worker to send a heartbeat from within a Job. I can't see any way of getting a reference to the current worker from an executing function though (I can get the current Job using `rq.get_current_job()` easily enough).

My use case is for very long running tasks (they typically take ~20 hours to complete) which do work in small chunks (typically each chunk completes in 5 minutes). Ideally, I'd like the task to have a reasonably short timeout which is refreshed by calling `Worker.heartbeat` each time a chunk completes.

At the moment, my timeout has to be >20 hours, which means it ends up being a long time before failures from e.g. hard shutdowns are noticed. The task has a very expensive set up, otherwise I'd consider splitting it into many smaller tasks in the first place.

**Comment by selwin** [comment]
I think we can attach a reference to `worker` in `job` instance so you can access it from `job`. Something like this:
```python
def prepare_job_execution(self, job):
    # Other code...
    job.worker = self
```

What do you think? If this works ok for you, please open a PR for this.

**Comment by davechallis** [comment]
Sounds great to me, will sort out a PR shortly.

**Comment by theodesp** [comment]
Can we close this ticket as this functionality has been merged with #824 ?

**Comment by selwin** [comment]
Fixed in https://github.com/rq/rq/pull/945
